### ISSUE TITLE
Remote print for voided/cancelled items - unexpected behavior

### ISSUE THREAD

**Issue by otc-narendra** [CORE-POS/IS4C#989]
* Version of CORE?
2.7
* Issue with Office, Lane, or both?
Lane
* Is this [mostly] a bug report, feature request, or question?
Question/bug

+ If you refund an item that’s included in the remoteprint table, the remote printer prints with a -1 qty. I would think that remote print should not print anything for refunded line items.

+ If you void an item that’s included in the remoteprint table, the remote printer prints a line for it with a 0 qty. Although I can imagine scenarios where telling the kitchen (via remote printed receipt) that an item has been cancelled subsequent to a suspend/resume cycle could be interesting for an open tab, I think it’s a bit confusing to figure out all the various usage scenarios. So, I would suggest that voided items don’t print on remote printer at all.

- Narendra

**Comment by gohanman** [comment]
That all sounds sensible to me. This should cover all the described oddities and edge cases:
dd8e9cc4aa6f2088c6b1e10c549e1dc8fdc4ddf7

**Comment by otc-narendra** [comment]
Hi Andy & Joel

I tried to integrate these changes to our installation but noticed that last Summer, Joel had made a bunch of modifications to our local copy of RemotePrint.php to fix some bugs and make a few other modifications. I have attached a copy of our version of the code to this email. I tried downloading the latest version of RemotePrint.php from the master branch with all your latest changes but it had some of the same problems that Joel fixed in our version including the fact that I cannot turn debug mode off and even if I just comment out that check (like it was in our version) the printing does not work properly.

I’m a git noob so not sure how you manage situations like this. Is there an Our-Table fork for the repo? What mechanism can I use to ensure that when I update my codebase with changes in the master branch that I don’t lose any customizations that Joel has done over the years?

Thanks,


  *   Narendra
